### PR TITLE
iOS 13: deviceorientation error for showCompass

### PR DESCRIPTION
**Pull request by sveine**

### CODE REVIEW

File: src/L.Control.Locate.js [hunk 1]
```diff
@@ -467,10 +467,27 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
...
-                        L.DomEvent.on(window, 'deviceorientation', this._onDeviceOrientation, this);
+                    var oriAbs = 'ondeviceorientationabsolute' in window;
+                    var oriStd = 'ondeviceorientation' in window;
+                    if (oriAbs || oriStd) {
+                        var _this = this;
+                        var deviceorientation = function () {
+                            if (oriAbs) {
+                                L.DomEvent.on(window, 'deviceorientationabsolute', _this._onDeviceOrientation, _this);
+                            } else if (oriStd) {
+                                L.DomEvent.on(window, 'deviceorientation', _this._onDeviceOrientation, _this);
```
Comment (domoritz) [inline src/L.Control.Locate.js]: Can you only set the second argument based on the oriAbs or oriStd?

File: src/L.Control.Locate.js [hunk 2]
```diff
@@ -467,10 +467,27 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
...
+                            }
+                        };
+                        if (DeviceOrientationEvent && typeof DeviceOrientationEvent.requestPermission === 'function') {
+                            DeviceOrientationEvent.requestPermission().then(function (permissionState) {
+                                deviceorientation();
+                                if (permissionState === 'granted') {
+                                    deviceorientation();
```
Comment (domoritz) [inline src/L.Control.Locate.js:479]: Why do we call the same function twice if granted is true?

**Review by domoritz — COMMENTED** [review]
Thank you. I have two comments.

File: src/L.Control.Locate.js (hunk 1 above)
Comment (sveine) [reply src/L.Control.Locate.js]: Agree!

File: src/L.Control.Locate.js (hunk 2 above)
Comment (sveine) [reply src/L.Control.Locate.js:479]: The statement was for debugging: removed

**Pull request merged**